src/main/java/uk/gov/hmcts/darts/courthouse/service/CourthouseServiceImpl.java

```diff
@@ -121,6 +135,150 @@ public CourthouseEntity addCourtHouse(Courthouse courthouse) {
+                                                              SecurityRoleEntity securityRoleRequester, SecurityRoleEntity securityRoleApprover) {
+        Set<SecurityGroupEntity> securityGroupEntities =
+            (CollectionUtils.isEmpty(mappedEntity.getSecurityGroups())) ? new LinkedHashSet<>() : mappedEntity.getSecurityGroups();
+
+        for (Integer id : securityRoleIds) {
+            SecurityGroupEntity securityGroupEntity = addSecurityGroupForCourthouse(courthousePost, getSecurityRoleByRoleId(id));
+            securityGroupEntities.add(securityGroupEntity);
+        }
+        SecurityGroupEntity securityGroupRequesterEntity = addSecurityGroupForCourthouse(courthousePost, securityRoleRequester);
+        SecurityGroupEntity securityGroupApprovarEntity = addSecurityGroupForCourthouse(courthousePost, securityRoleApprover);
+
+        securityGroupEntities.add(securityGroupApprovarEntity);
+        securityGroupEntities.add(securityGroupRequesterEntity);
+        return securityGroupEntities;
+    }
+
+    private SecurityGroupEntity addSecurityGroupForCourthouse(CourthousePost courthousePost, SecurityRoleEntity securityRole) {
+        SecurityGroupEntity securityGroupEntity = new SecurityGroupEntity();
+
+        securityGroupEntity.setDisplayName(courthousePost.getCourthouseName());
+        securityGroupEntity.setGroupName(courthousePost.getCourthouseName());
+        securityGroupEntity.setGlobalAccess(false);
+        securityGroupEntity.setDisplayState(true);
+        securityGroupEntity.setUseInterpreter(false);
+        securityGroupEntity.setSecurityRoleEntity(securityRole);
+        securityGroupRepository.saveAndFlush(securityGroupEntity);
+
+        return securityGroupEntity;
+    }
+
+    private SecurityRoleEntity getSecurityRoleByRoleName(String scurityRole) {
+        assert securityRoleRepository != null;
+        List<SecurityRoleEntity> securityRoleEntities = securityRoleRepository.findAllByOrderById();
+        for (SecurityRoleEntity securityRoleEntity: securityRoleEntities) {
+            if (securityRoleEntity.getRoleName().equals(scurityRole)) {
+                return securityRoleEntity;
+            }
+        }
+        return null;
+    }
+
+    private SecurityRoleEntity getSecurityRoleByRoleId(Integer roleId) {
+        List<SecurityRoleEntity> securityRoleEntities = securityRoleRepository.findAllByOrderById();
+        for (SecurityRoleEntity securityRoleEntity: securityRoleEntities) {
+            if (securityRoleEntity.getId().equals(roleId)) {
+                return securityRoleEntity;
+            }
+        }
+        return null;
+    }
+
+    private void checkCourthouseDisplayNameIsUnique(CourthousePost courthousePost) {
+        Optional<CourthouseEntity> foundCourthouse = courthouseRepository.findByCourthouseNameIgnoreCase(courthousePost.getDisplayName());
```

Comment:
This one's doing the incorrect lookup, should be calling a `findByDisplayNameIgnoreCase()` repository method.